rpm/SOURCES/cntlm.service

```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=cNTLM authentication Proxy
+After=network.target
+
+[Service]
+Type=simple
+Environment="RUNAS=cntlm"
+EnvironmentFile=-/etc/sysconfig/cntlmd
+ExecStart=/usr/sbin/cntlm -f -U $RUNAS -c /etc/cntlm.conf $OPTARGS
+
+[Install]
+WantedBy=multi-user.target
```

Comment:
Valid point; the above is consistent with how this service is managed as a daemon in the FedoraProject systemd implementation; however you make a very valid point.

What would be your thoughts on a parameterized SD unit that allows running an instance per user @jschwartzenberg ? 

This approach of system level, parameterized per-user, systemd units is also the approach that I see a lot of similar per-user network services like code-server and tigervnc taking as well.

Reply:
Cntlm has also the gateway mode, that is it can be accessed from other computers. It implements also access control lists. So it makes sense to be installed system wide.

Reply:
> What would be your thoughts on a parameterized SD unit that allows running an instance per user @jschwartzenberg ?

That would be perfect. Out of the box it would not be enabled and in the documentation there would be steps to enable it either per user or system-wide.

I remember there was also a project from the systemd corner that would provide different users with their own loopback interface through namespaces. With such a set-up, even when multiple users are logged in on the same system, their connections would always use their associated credentials with each user running their own copy of cntlm.

Reply:
Sounds reasonable; therefore will get on implementing and testing this.

Reply:
Implemented an initial draft of this in [fd484c4](https://github.com/versat/cntlm/pull/97/commits/fd484c4b4d72dc2ac9f5b6185e74ab1abba34891) 

Will document this accordingly (provided this looks alright implementation wise) next week.

Reply:
Apologies for the late response. I added some remarks that might help. I guess a good example to look at work be for instance PulseAudio which is also running per user.

Reply:
I was mainly thinking of running per-user but still at the system scope in the context of systemd as not all distros fully support user scoped units.

However as long as there is not an aversion to being rather progressive, I can refactor to be a true user scoped unit akin to PA.

Reply:
Given the above, @jschwartzenberg let me know if you have strong feelings in favor of a systemd system scope service running as a normal user (more compatible) or a true user unit (more "proper", however not well supported on anything older than RHEL/centos 7).

Reply:
I would not expect any systems older than RHEL 7 to be using systemd at all if they are still in use. RHEL 6 didn't use systemd yet. My proposal would be to do it the proper way. For older systems the init file could be kept around, but I wouldn't recommend spending much time on it. If anybody wants to support an old system or another init system, they may create their own PR.

Of course I don't know your situation, be sure that whatever you do is also catering towards that :)
In my situation this is really about RHEL 8 and higher. Ideally the configuration would be managed at the system-wide level at least with regards to where the PAC file is coming from. The credentials are handled through Kerberos per user.

In the perfect situation there would be an separate loopback interface, private for each user, on which cntlm could always bind such that each user would have their own instance running with their own credentials without disturbing or exposing other instances. I imagine @poettering may have a brilliant idea how that had best be achieved. I thought I saw an article in the past about having a loopback interface private for each user through systemd, but I can't find it anymore. Of course that would seem overkill for just this PR. Right now if one user logs in, their cntlm copy would claim the port. An instance of another user cannot claim the same port and the port of the instance with the first user's credentials will be exposed to other users logging in as they have full access to the loopback interface.

Reply:
Legacy support is not a major concern for us as well, therefore that answers the question with regards to parameterized system scoped units as opposed proper user units; proper user units 100%.

However I would like to have the option of being able to run either per-user (where each user uses credentials from a configuration located in their home directory) or a single systemwide instance (in cases where for example there is a single AD backed machine service account that all processes on a multi-user system share for proxy auth). As a result, my personal preference would be to have a traditional system scoped systemd unit along with a native user unit that takes configuration from a user's home directory.

To address @jschwartzenberg 's ask to be able to manage the PAC file system-wide, my proposal here is to make both the user as well as system scoped units append arguments specified in the /etc/sysconfig/cntlmd OPTARGS variable to the command line of cntlm. This would allow you to enforce that all PAC files are taken from a specific location that is managed system wide.

To the point about private loopback interfaces; I think you are referring to network namespaces as those would provide exactly the type of functionality that you are referring to; however under the constraint that the process connecting as well as the process listening are both part of the same netns.

Reply:
implemented the above in feb5349bcb5a963ac17366793adbbfb2fa254e30